controllers/pravegacluster_controller.go

```diff
@@ -388,11 +388,28 @@ func (r *PravegaClusterReconciler) reconcileControllerService(p *pravegav1beta1.
 
 	service := MakeControllerService(p)
 	controllerutil.SetControllerReference(p, service, r.Scheme)
-	err = r.Client.Create(context.TODO(), service)
-	if err != nil && !errors.IsAlreadyExists(err) {
-		return err
-	}
+	currentService := &corev1.Service{}
+	err = r.Client.Get(context.TODO(), types.NamespacedName{Name: service.Name, Namespace: p.Namespace}, currentService)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			err = r.Client.Create(context.TODO(), service)
+			if err != nil && !errors.IsAlreadyExists(err) {
+				return err
+			}
 
```

Comment:
This line still looks strange

Reply:
here  are you  suggesting to remove the `IsAlreadyExists` check?

Reply:
I mean the empty line

Reply:
Done